src/mesh_tools.rs

```diff
@@ -150,49 +150,36 @@ pub fn separate_isolated_meshes(geometry: &Geometry) -> Vec<Geometry> {
     let mut available_face_indices: HashSet<u32> = face_to_face.keys().copied().collect();
     let mut patches: Vec<Geometry> = Vec::new();
 
-    while let Some(first_face_index) = available_face_indices.iter().next() {
-        let connected_indices = crawl_faces(*first_face_index, &face_to_face);
+    while let Some(start_face_index) = available_face_indices.iter().next() {
+        let mut index_stack = vec![*start_face_index];
+        let mut connected_face_indices = HashSet::new();
+
+        while let Some(current_face_index) = index_stack.pop() {
+            if connected_face_indices.insert(current_face_index) {
+                available_face_indices.remove(&current_face_index);
```

Comment:
As for your cry for help, you can use `available_face_indices.iter().cloned().next()` to please the borrow checker. This shouldn't even add any significant overhead in case there's just couple of different meshes.

Reply:
great, that works! thanks